docs/guides/cluster-to-advanced-cluster-migration-guide.md

```diff
@@ -0,0 +1,181 @@
+# Cluster to Advanced Cluster Migration Guide
+
+**Objective**: Guide users to replace the `mongodbatlas_cluster` resource with the `mongodbatlas_advanced_cluster` resource.
+
+**Note**: This guide focus on the resource migration as the data source migration is only requiring a resource_type change from `data.mongodbatlas_cluster` to `data.mongodbatlas_advanced_cluster`.  However, pay attention to the [output changes.](#output-changes)
+
+## Main Changes Between `mongodbatlas_cluster` and `mongodbatlas_advanced_cluster`
+
+Key differences in the configuration:
+
+1. Replication Spec Configuration: Support different node types (electable, analytics, read_only) where hardware configuration can differ between node types.
+2. Provider Settings: Moved from the top level to the replication spec allowing you to create multi-cloud clusters.
+3. Auto Scaling: Moved from the top level to the replication spec allowing you to scale replication specs individually.
+4. Backup Configuration: Renamed from `cloud_backup` to `backup_enabled`.
+
+### Example1: Old Configuration
+
+```terraform
+resource "mongodbatlas_cluster" "this" {
+    project_id                   = var.project_id
+    name                         = "legacy-cluster"
+    cluster_type                 = "REPLICASET"
+
+    # Provider Settings "block"
+    provider_instance_size_name = "M10" # 1
```

Comment:
i'd add something else in the comment so you don't need to go up and down to see what they were, like:
# 1 rep spec
# 2 provider
...